frontend/src/pages/UnitDetails.tsx

```diff
@@ -290,6 +333,46 @@ export function UnitDetails() {
   //checks for which view to return
   const [isEditing, setIsEditing] = useState(false);
 
+  const [imgUrls, setImgUrls] = useState<string[]>([]);
+  const [vidUrls, setVidUrls] = useState<string[]>([]);
+
+  const getFiles = () => {
+    listAll(ref(storage, `${id}/images/`))
```

Comment:
I noticed that the code for getting images/videos is duplicated in a few places. What are your thoughts on extracting that functionality out as a separate service (e.g. in `src/api/images.ts` or something)?